File: pgm/ABC_Dart/ABC_332_A.dart
//  'dart:collection';
import 'dart:io';
// import 'dart:math';
import 'dart:typed_data';

void main() {
  final [int N, int S, int K] = stdin
      .readLineSync()!
      .split(' ')
      .map(int.parse)
      .toList();
  const String space = ' ';
  const int one = 1;
  const int zero = 0;
  final Int32List P = Int32List(N);
  final Int32List Q = Int32List(N);
  for (int i = zero; i < N; i++) {
    final List<int> arr = stdin
        .readLineSync()!
        .split(space)
        .map(int.parse)
        .toList();
    P[i] = arr[zero];
    Q[i] = arr[one];
  }
  int price = zero;
  for (int i = zero; i < N; i++) {
    price += P[i] * Q[i];
  }
  if (price < S) {
    print(price + K);
  } else {
    print(price);
  }
  exit(0);
}

File: pgm/ABC_Kotlin/ABC_332_A.kt
// import java.io.BufferedOutputStream
// import java.io.BufferedReader
// import java.io.DataInputStream
// import java.io.FileInputStream
// import java.io.IOException
// import java.io.InputStreamReader
// import java.io.PrintWriter
// import java.lang.StringBuilder
// import java.time.*
// import java.util.StringTokenizer

// import kotlin.math.*

const val space: String = " "
const val one: Int = 1
const val zero: Int = 0

fun main() {
    val (N: Int, S: Int, K: Int) = readln().split(space).map { it.toInt() }
    val P: IntArray = IntArray(N)
    val Q: IntArray = IntArray(N)
    for (i in zero until N) {
        val arr: List<Int> = readln().split(space).map { it.toInt() }
        P[i] = arr[zero]
        Q[i] = arr[one]
    }
    var price: Int = zero
    for (i in zero until N) {
        price += P[i] * Q[i]
    }
    if (price < S) {
        println(price + K)
    } else {
        println(price)
    }
    kotlin.system.exitProcess(0)
}